common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -718,7 +723,7 @@ def _load_items(self, course_entry, block_keys, depth=0, lazy=True, **kwargs):
         if runtime is None:
             runtime = self.create_runtime(course_entry, lazy)
             self._add_cache(course_entry.structure['_id'], runtime)
-            self.cache_items(runtime, block_keys, course_entry.course_key, depth, lazy)
+            self.cache_items(runtime, block_keys, course_entry.course_key, depth, depth == 0 and lazy)
```

Comment:
My hesitation to do so is based on not knowing the full implications on performance -and- correctness. I guess correctness should be covered by unittests. But since we don't have any regular performance testing, it's difficult for me to say if any part of the system is relying on being able to query course structure definitions using `depth=0` and `lazy=True`.

@cpennington Do you see a downside to me making this change? 

I'll make the change and let the unittests run. (And add a comment describing the intent.)